arch/riscv/core/smp.c

```diff
@@ -74,7 +74,11 @@ void arch_secondary_cpu_init(int hartid)
 #endif
 #ifdef CONFIG_SMP
 	irq_enable(RISCV_IRQ_MSOFT);
-#endif
+#ifdef CONFIG_RISCV_PLIC_HAS_IRQ_AFFINITY
+	/* Enable on secondary cores so that they can respond to PLIC */
+	irq_enable(RISCV_IRQ_MEXT);
```

Comment:
It's somewhat surprising that this has been missing for so long 😅

I guess it would be a non-trivial improvement?

Reply:
> I guess it would mean a non-trivial improvement?

That's yet to be found out, although every core can handle the IRQ after this patch, only 1 core would be able to actually claim it while the others simply waste CPU cycles. 

I guess we will need a new benchmark to test the overall interrupt handling performance

Reply:
I would say a couple of key things to verify:

1. If an interrupt is pinned only to certain harts / cores, it would be ideal if other cores were simply not interrupted at all.

2. If an interrupt is masked / available to multiple cores, and if the arch must interrupt more than a single core, then the cores that did not receive the interrupt should more or less immediately bounce directly out of the ISR. Maybe some intc's are capable of only selecting / interrupting a single core (that would be more ideal than contention).

Probably would be good to craft a testsuite.

Also perf analyses before and after the change to see if a) there is significant cpu time lost, and b) to measure interrupt latency in some kind of interrupt storm